R/class.R

```diff
@@ -2,21 +2,37 @@ inject_super <- function(fun) {
   # for each function `fun` we need to place a `super` function in its
   # search path, and this `super` function must be able to access
   # the `self` argument passed to `fun`.
-  
+
   e <- new.env(parent = environment(fun))
   
   e$super <- function() {
     bt <- reticulate::import_builtins()
     # self is an argument passed to fun
-    self <- get("self", envir = parent.frame(), inherits = FALSE) 
-    bt$super(self$`__class__`, self)
+    self <- get("self", envir = parent.frame(), inherits = FALSE)
+    class_ <- get("__class__", envir = e, inherits = FALSE)
+    bt$super(class_, self)
   }
-  
+
   environment(fun) <- e # so fun can access `super`
-  
+
   fun
 }
 
+#' Enable the convertion scope so `self` fields can be accessed
+#' without the need to call `py_to_r`.
+#' 
+#' @param f a method/function of a Python class
+#' 
+enable_convert_scope <- function(f) {
```

Comment:
The name of the function implies it's more generic than it really is (it's a very narrow helper used explicitly for PyClass). I'd prefer to just keep it a private helper since I doubt it would be reused in any other context.